skiko/build.gradle.kts

```diff
@@ -573,74 +575,179 @@ fun skiaStaticLibraries(skiaDir: String, targetString: String): List<String> {
+                "-framework", "CoreGraphics",
+                "-framework", "CoreServices",
+                "-framework", "CoreText",
+                "-framework", "Foundation",
+                "-framework", "IOKit",
+                "-framework", "Metal",
+                "-framework", "OpenGL",
+                "-framework", "QuartzCore" // for CoreAnimation
             )
         }
         OS.Linux -> {
-            compilerArgs.addAll(
-                listOf(
-                    "-fno-rtti",
-                    "-fno-exceptions",
-                    "-fvisibility=hidden",
-                    "-fvisibility-inlines-hidden",
-                    "-I$jdkHome/include/linux",
-                    "-DSK_BUILD_FOR_LINUX",
-                    "-D_GLIBCXX_USE_CXX11_ABI=0",
-                    *buildType.clangFlags
-                )
+            osFlags = arrayOf(
+                "-shared",
+                "-static-libstdc++",
+                "-static-libgcc",
+                "-lGL",
+                "-lfontconfig",
+                // A fix for https://github.com/JetBrains/compose-jb/issues/413.
+                // Dynamic position independent linking uses PLT thunks relying on jump targets in GOT (Global Offsets Table).
+                // GOT entries marked as (for example) R_X86_64_JUMP_SLOT in the relocation table. So, if there's code loading
+                // platform libstdc++.so, lazy resolve code will resolve GOT entries to platform libstdc++.so on first invocation,
+                // and so further execution will break, as those two libstdc++ are not compatible.
+                // To fix it we enforce resolve of all GOT entries at library load time, and make it read-only afterwards.
+                "-Wl,-z,relro,-z,now",
+                // Hack to fix problem with linker not always finding certain declarations.
+                "$skiaBinDir/libsksg.a",
+                "$skiaBinDir/libskia.a",
+                "$skiaBinDir/libskunicode.a"
             )
         }
         OS.Windows -> {
-            compilerArgs.addAll(
-                listOf(
-                    "-I$jdkHome/include/win32",
-                    "-DSK_BUILD_FOR_WIN",
-                    "-D_CRT_SECURE_NO_WARNINGS",
-                    "-D_HAS_EXCEPTIONS=0",
-                    "-DWIN32_LEAN_AND_MEAN",
-                    "-DNOMINMAX",
-                    "-DSK_GAMMA_APPLY_TO_A8",
-                    "-DSK_DIRECT3D",
-                    "/utf-8",
-                    "/GR-", // no-RTTI.
-                    // LATER. Ange rendering arguments:
-                    // "-I$skiaDir/third_party/externals/angle2/include",
-                    // "-I$skiaDir/src/gpu",
-                    // "-DSK_ANGLE",
-                    *buildType.msvcFlags
-                    )
+            linker.set(windowsSdkPaths.linker.absolutePath)
+            libDirs.set(windowsSdkPaths.libDirs)
+            osFlags = arrayOf(
+                "/DEBUG",
```

Comment:
`"/DEBUG"` shall exist in debug builds only, in release it shall be optimization flags.